### PR TITLE
docs(root): add empty state documentation

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes

Add empty state documentation with code examples. To test this, the components will have to be built locally from develop until the next release. 

## Related issue

 #487

## Checklist

- [x] I have [manually accessibility tested](https://design.sis.gov.uk/accessibility/testing/manual-testing) any changes, if relevant.
- [x] I have raised this pull request against the [develop branch](https://github.com/mi6/ic-design-system/tree/develop)

### CODE REVIEW

File: src/content/structured/components/empty-state/accessibility.mdx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+path: "/components/empty-state/accessibility"
+
+date: "2023-08-08"
+
+title: "Empty State"
```
Comment (MI6-94) [inline src/content/structured/components/empty-state/accessibility.mdx:6]:
```suggestion
title: "Empty state"
```

File: src/content/structured/components/empty-state/code.mdx [hunk 2]
```diff
@@ -0,0 +1,424 @@
+---
+path: "/components/empty-state/code"
+
+date: "2023-08-09"
+
+title: "Empty State"
```
Comment (MI6-94) [inline src/content/structured/components/empty-state/code.mdx:6]:
```suggestion
title: "Empty state"
```

File: src/content/structured/components/empty-state/guidance.mdx [hunk 3]
```diff
@@ -0,0 +1,219 @@
+---
+path: "/components/empty-state"
+
+navPriority: 11
+
+date: "2023-08-09"
+
+title: "Empty State"
```
Comment (MI6-94) [inline src/content/structured/components/empty-state/guidance.mdx:8]:
```suggestion
title: "Empty state"
```

File: src/content/structured/components/empty-state/guidance.mdx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+  </IcEmptyState>
+</ComponentPreview>
+
+## When to use
+
+Use an empty state component to communicate to people that no content is available to display. Use them to engage people and guide them on what is best to do next by including a button or a link to the next best action.
+
+Empty states are used for many different reasons such as:
+
+- No content yet exists due to the first-time use of the functionality,
```
Comment (MI6-94) [inline src/content/structured/components/empty-state/guidance.mdx:76]:
```suggestion
- No content yet exists due to the first-time use of the functionality.
```

File: src/content/structured/components/empty-state/guidance.mdx [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+- Data was not successfully loaded,
```
Comment (MI6-94) [inline src/content/structured/components/empty-state/guidance.mdx:77]:
```suggestion
- Data was not successfully loaded.
```

File: src/content/structured/components/empty-state/guidance.mdx [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+- A query or search term yielded no results,
```
Comment (MI6-94) [inline src/content/structured/components/empty-state/guidance.mdx:78]:
```suggestion
- A query or search term yielded no results.
```

File: src/content/structured/components/empty-state/guidance.mdx [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+Use empty states when no data can be displayed in a component’s contents. For example, show an empty state when no data is available to display in a data table, or if a search term didn’t find any results.
+
+<DoDontCaution
+  imageSrc={emptyStateFig1}
+  imageAlt="An example app called ‘We love coffee’ with a page displaying favourite teas. An empty data table is displayed with an empty state that reads ‘There’s nothing here’."
+  state="good"
+  caption="Use empty states when there is no data available to display."
+/>
+
+Use empty states to engage people and provide guidance on what is best to do next.
```
Comment (MI6-94) [inline src/content/structured/components/empty-state/guidance.mdx:90]: Remove? Repeat of above?

File: src/content/structured/components/empty-state/guidance.mdx [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+
+<DoDontCaution imageSrc={emptyStateFig12} imageAlt="" state="bad" caption="" />
+
+## Accessibility considerations
+
+If an empty state is displayed as a result of a change on a page, make sure it is announced by a screen reader using a live region.
+
+## Related components
+
+- [Alert](/components/alert)
```
Comment (MI6-94) [inline src/content/structured/components/empty-state/guidance.mdx:218]:
```suggestion
- [Alert](/components/alert).
```

File: src/content/structured/components/empty-state/guidance.mdx [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+- [Loading indicator](/components/loading-indicator)
```
Comment (MI6-94) [inline src/content/structured/components/empty-state/guidance.mdx:219]:
```suggestion
- [Loading indicator](/components/loading-indicator).
```

File: src/content/structured/components/empty-state/guidance.mdx (hunk 7 above)
Comment (DBD324) [reply src/content/structured/components/empty-state/guidance.mdx]: Good catch

File: src/content/structured/components/empty-state/guidance.mdx [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+import emptyStateFig5 from "./images/fig-5-dont-use-empty-states-for-messages-unrelated-to-empty-state.png";
+import emptyStateFig6 from "./images/fig-6-dont-use-empty-state-for-loading-messages.png";
+import emptyStateFig7 from "./images/fig-7-use-empty-states-on-a-full-screen.png";
+import emptyStateFig8 from "./images/fig-8-use-empty-states-on-a-side-panel.png";
+import emptyStateFig9 from "./images/fig-9-use-empty-states-on-other-containers.png";
+import emptyStateFig10 from "./images/fig-10-empty-state-anatomy.png";
+import emptyStateFig11 from "./images/fig-11-avoid-using-too-many-actions.png";
+import emptyStateFig12 from "./images/fig-12-dont-use-other-component-in-interaction-area.png";
+
+## Introduction
```
Comment (DBD324) [inline src/content/structured/components/empty-state/guidance.mdx:38]: Remove heading of introduction and replace with heading 'Component demo'

File: src/content/structured/components/empty-state/guidance.mdx [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+
+An example of the empty state component.
```
Comment (DBD324) [inline src/content/structured/components/empty-state/guidance.mdx:40]: remove as unnecessary

**Review by DBD324 — CHANGES_REQUESTED** [review]
A couple of changes requested. Need to wait for build before approval can be given. Need to see page to check sizing of imagery etc.

**Comment by DBD324** [comment]
When reviewing images noticed two required changes:
- Add warning status to the second image in the content section. 
- Reduce size of second image in when not to use section to a half width image. 

Otherwise looks good.

**Review by DBD324 — APPROVED** [review]
All comments have been actioned. Guidance is looking good to me.

**Review by MI6-255 — APPROVED** [review]

**Comment by MI6-255** [comment]
Approved, just need the build fixing

**Pull request merged**